### ISSUE TITLE
An option to use separate dirs

### ISSUE THREAD

**Issue by DarthGandalf** [sergiou87/open-supaplex#7]
I'm trying to package this for my linux distro, and various resources belong to `/usr/share`, but files like `PLAYER.LST`, `HALLFAME.LST` should be somewhere in `$HOME`, because `/usr/share` is not writeable by the user. The binary itself is in yet another dir: `/usr/bin/opensupaplex`

Probably resource location could even be hardcoded as a compile-time option, default to `.`, and I'd override it to `/usr/share/opensupaplex`

Saves location can't be hardcoded like that… I don't have a good suggestion there. Maybe provide a command-line flag, also defaulting to `.`?

WDYT?

**Comment by sergiou87** [comment]
Yeah I don't know what's the best option either 🤔  If you check this https://github.com/sergiou87/open-supaplex/blob/3f7708b6ac75e255a9e429e3cf5aed335e116a7e/src/file.c#L60-L70 and https://github.com/sergiou87/open-supaplex/blob/7b62e93f711b58925968d34467a2e27a8a9ed7e9/src/sdl_common/audio.c#L73-L85 there are a few ways to set specific paths per platform.

The "writable" files are those which are changed by the game. Some platforms (like PS Vita) can store resources with the game binary, but then can't write them, so the game creates a specific (hardcoded) path somewhere else where those files you mentioned can be written. For that I added this: https://github.com/sergiou87/open-supaplex/blob/3f7708b6ac75e255a9e429e3cf5aed335e116a7e/src/file.c#L27

So in Linux we might need to do something like that, and just create a `getWritableFilePath` that points to `~/.config/OpenSupaplex`? Does that sound reasonable?

**Comment by DarthGandalf** [comment]
Only ADVANCED.CFG seems to be reading via `getWritableFilePath`, and even that file it tries to write it to `getReadonlyFilePath` back.

**Comment by DarthGandalf** [comment]
And there need to be some way to select whether to use files in `.` or in those 2 directories. Probably a compile-time option would suffice for that though

**Comment by DarthGandalf** [comment]
> Only ADVANCED.CFG seems to be reading via getWritableFilePath, and even that file it tries to write it to getReadonlyFilePath back.

Nvm, that's me putting `#endif` to the wrong line

**Comment by sergiou87** [comment]
> And there need to be some way to select whether to use files in `.` or in those 2 directories. Probably a compile-time option would suffice for that though

Not super proud of what I'm gonna suggest, but why not hardcoding `./` for Linux for readonly files? And then something like `~/.config/OpenSupaplex` for writable files. It can be extracted to a compile-time param later if needed (in case some distro/package format needs it).

It's a pity how different platforms required different solutions to access the game files 😢
